packages/connect/src/connector/smoldot-light.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import type {
+  Chain as SChain,
+  Client,
+  ClientOptions,
+  AlreadyDestroyedError as IAlreadyDestroyedError,
+  CrashError as ICrashError,
+  JsonRpcDisabledError as IJsonRpcDisabledError,
+} from "@substrate/smoldot-light"
+import {
+  AlreadyDestroyedError,
+  CrashError,
+  JsonRpcDisabledError,
+} from "./errors.js"
+import { getSpec } from "./specs/index.js"
+import type {
+  AddChain,
+  AddWellKnownChain,
+  Chain,
+  SubstrateConnector,
+} from "./types.js"
+import { SupportedChains } from "../SupportedChains.js"
+
+let SdAlreadyDestroyedError: typeof IAlreadyDestroyedError
+let SdCrashError: typeof ICrashError
+let SdJsonRpcDisabledError: typeof IJsonRpcDisabledError
+
+let startPromise: Promise<(options: ClientOptions) => Client> | null = null
+const getStart = () => {
```

Comment:
Why are `start` and `startPromise` separate from `clientPromise` and `getClient`?

Reply:
The `startPromise` either gets resolved once (the Extension is not present), or it never gets resolved at all. The `clientPromise` on the other hand will exist as long as there is one active client, once the client gets terminated, then `clientPromise` gets set to `null`, so that if in the future a new client needs to be created it doesn't try to access the terminated one.

Reply:
I was thinking that browsers are probably smart enough to not actually import smoldot twice if you run `getStart` twice, but yeah it doesn't hurt to explicitly only ever do it once.